Commencement.Mvc/Controllers/MajorCodeController.cs

```diff
@@ -104,14 +106,27 @@ public ActionResult Add(string majorId, string majorCode, string majorName)
 
             if (ModelState.IsValid)
             {
+                major.TransferValidationMessagesTo(ModelState);
+                if (major.Id.Trim().Length > 4)
+                {
+                    ModelState.AddModelError("MajorCode.Id", "Major Code has a max length of 4 characters.");
+                }
+            }
+
+            if (ModelState.IsValid)

```

Comment:
@jSylvestre this will work but we don't really need multiple model state checks here, we can just do the checks and transfer validation messages no matter what i think.